### DISCUSSION TITLE
What should I expect using `--files-without-match` and `--count`

### DISCUSSION THREAD

**Discussion by FrancescElies** [BurntSushi/ripgrep#2404] [Q&A]
If I want to search files without a certain word I would do as follows
```
ripgrep on  master [⇡] is 📦 v13.0.0 via 🦀 v1.66.0
❯ rg --files-without-match "strict" -tmd
RELEASE-CHECKLIST.md
README.md
crates\searcher\README.md
crates\regex\README.md
crates\printer\README.md
crates\pcre2\README.md
crates\matcher\README.md
crates\ignore\README.md
crates\grep\README.md
crates\globset\README.md
crates\core\README.md
crates\cli\README.md
ci\docker\README.md
benchsuite\runs\2022-12-16-archlinux-duff\README.md
benchsuite\runs\2020-10-14-archlinux-frink\README.md
.github\ISSUE_TEMPLATE\feature_request.md
.github\ISSUE_TEMPLATE\bug_report.md
```
Now if I want to count how many files don't contain a certain match I would expect the following to give me that answer but instead it is the same as if I search for files matching
```
ripgrep on  master [⇡] is 📦 v13.0.0 via 🦀 v1.66.0
❯ rg --files-without-match "strict" -tmd -c
GUIDE.md:6
FAQ.md:1
CHANGELOG.md:3

ripgrep on  master [⇡] is 📦 v13.0.0 via 🦀 v1.66.0
❯ rg "strict" -tmd -c
GUIDE.md:6
FAQ.md:1
CHANGELOG.md:3
```

Are my expectations of `rg --files-without-match "strict" -tmd -c` wrong?
```
ripgrep on  master [⇡] is 📦 v13.0.0 via 🦀 v1.66.0
❯ rg --version
ripgrep 13.0.0
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)
```

**Comment by BurntSushi** [comment]
I think this might be a documentation bug (and a partial implementation hiccup). Namely, the code implements `--files-without-match` and `--count` as mutually exclusive options:

https://github.com/BurntSushi/ripgrep/blob/fe97c0a152cabc1bc07ec36b4b1e27cd230c3014/crates/core/args.rs#L1585-L1600

But the actual argument parser doesn't define `--files-without-match` and `--count` as flags that override one another. That means both `--files-without-match --count` and `--count --files-without-match` behave the same, but the former should show match counts while the latter should show all files that don't match.

The reason why ripgrep behaves this way is because it doesn't make sense to combine `--files-without-match` with `--count`. There's nothing to count. The behavior is merely to show any file that does not match.

Now there is the related query of "show me a count of the number of lines that do not match my query." You can do that with `--files-with-matches --count --invert-match`, or `-lcv` for short.
